### PR TITLE
Disable one of the GPT2 SDPA patterns for single-thread case

### PR DESCRIPTION
**Pull request by sanchitintel**
In #123503, it was reported that for BF16 dtype, accuracy is poor when only one thread is used for inference on CPU & Inductor uses one of the two SDPA patterns for GPT2 to offload compute to `torch.nn.functional.scaled_dot_product_attention`.
Couldn't find its root-cause, so disabling this pattern for now, when only one thread is used (which probably isn't done in practice, except in some benchmarks).
This issue isn't observed with multiple threads, or for FP32 datatype. It's not observed on Intel Xeon Ice Lake SP machines either, and is specific to a single thread config with processor supporting at least AVX512_BF16 ISA. Its root-cause is not known at this point. Digging.

Thanks!

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by leslie-fang-intel — APPROVED** [review]
Can we also add the test case for single-thread case of `_sfdp_pattern_18`?

**Comment by sanchitintel** [comment]
Hi @leslie-fang-intel, the failure occurred with BF16 datatype on a Intel Xeon 4th gen machine, which has AVX512_BF16 & AMX_BF16 ISAs. Without these ISAs, on current PyTorch CI machines (Ice Lake SP or Eager Lake SP), BF16 would be converted to FP32 before computation.

Xeon 4th generation CI machines were recently added but it seems no CI jobs are currently running on them. However, #127758 would enable CPU AMP CI on those machines.

**Comment by sanchitintel** [comment]
Found root-cause
